L1Trigger/L1TGlobal/src/AXOL1TLCondition.cc

```diff
@@ -90,10 +90,25 @@ const bool l1t::AXOL1TLCondition::evaluateCondition(const int bxEval) const {
   int useBx = bxEval + m_gtAXOL1TLTemplate->condRelativeBx();
 
   //HLS4ML stuff
-  std::string AXOL1TLmodelversion = m_AXOL1TLmodelversion;  //config loading method
+  std::string AXOL1TLmodelversion = m_AXOL1TLmodelversion;  //loading from menu
+
+  //if model version is empty, throw exception. Should not ever happen unless not found in menu
+  if (AXOL1TLmodelversion == "" || AXOL1TLmodelversion == "GTADModel_") {
+    throw cms::Exception("ModelError") << " Error model version not found in menu!";
+  }
+
+  //otherwise load model (if possible) and run inference
   hls4mlEmulator::ModelLoader loader(AXOL1TLmodelversion);
   std::shared_ptr<hls4mlEmulator::Model> model;
-  model = loader.load_model();
+
+  //model = loader.load_model();
+  try {
+    model = loader.load_model();
+  } catch (std::runtime_error& e) {
+    edm::LogWarning("AXOL1TLCondition") << "ERROR: failed to load model version " << AXOL1TLmodelversion
+                                        << ". Not evaluating condition!" << std::endl;
+    return false;
```

Comment:
Sorry, I understood  this should also be a ```throw```  exception, rather than ```try/catch``` / returning false?

Reply:
This is, at this point, the last piece of fail-safe introduced to protect HLT's stability and operation in case anything got out of synch or was mismatched or AXOL1TL introduced a new version somewhere. We figured mismatching was preferable to crashing HLT until the discrepancy could be resolved. Do you want this removed?

@eyigitba @quinnanm (and this should be brought up to the AXO team) If this is removed, we should be clear that this likely restricts AXOL1TL updates to an HLT release schedule.

Reply:
How would this error arise? If it is due to a configuration issue then it should be caught by L1T/TSG testing before deployment and the config issue fixed before deplyment. Or could it somehow happen suddenly during data taking?

Reply:
No, the issue that I am envisioning is that the AXOL1TL team comes forward and says "we have these model updates that are otherwise minor to the point of transparency" for tweaking inputs or PU or whatever else in the middle of a run based on the run conditions. My fear is that in this scenario if we just proceed with the update from our side, the menu change implied to a new model version (which will key the emulator to look for new models) is now a full crash at HLT unless HLT updates as well to versions of CMSSW that have the new AXOL1TL in CMSSW. As far as I was aware HLT updated on a regular, slow schedule, but AXOL1TL may request to update differently.

Perhaps my understanding is off here however, does HLT have to perform full version updates anyways if we change the menu?

Reply:
HLT certainly does not want L1T changes under the hood - we envisage that any AXO update for production is handled like any other L1T menu update, eg, new version v1_0_1 => v1_0_2 etc.

Reply:
> My fear is that in this scenario if we just proceed with the update from our side, the menu change implied to a new model version

are you implying you could update the model without updating the L1T menu? Isn't this in contradiction with reading the model from the menu in the first place? If you change the menu we certainly need to test the HLT first!

Reply:
No, the menu and model changes would happen in synch, my point is that the AXOL1TL team may request these updates at a different frequency/schedule than before, even with everything otherwise the same. 

So: 

Is HLT okay with that? Can and would HLT be willing to stick to AXOL1TL's update schedule? 

Or, does AXOL1TL now need to stick to HLT's schedule, implying restrictions on what they can update when? I don't think this is something AXOL1TL ever considered for updating their own model, their team may not be aware of it, and may not agree to these restrictions.

Or are there other safeguards we may want to try to employ?

Reply:
For each new HLT menu version (1-5 in 2024) we also have the opportunity to update the L1T menu, but more than that is currently not foreseen, so would require discussion and agreement between TSG and L1T,

Reply:
@aloeliger , this is something I briefly discussed with AXO team and TSG yesterday/today. For now, due to tying this setting to the L1 menu, we will have to follow the usual L1+HLT menu deployment schedule. We can revisit this in the future, but current plan is to request a menu version update for every AXO update. For CICADA this can be different since its O2O will not be handled with the menu version.